app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -141,6 +142,9 @@ protected void onNewIntent(Intent unsafeIntent) {
             // We can't update our fragment right now because we need to wait until the activity is
             // resumed. So just remember this URL and load it in onResumeFragments().
             pendingUrl = intent.getDataString();
+            if (intent.hasExtra(HomeScreen.ADD_TO_HOMESCREEN_TAG)) {
+                TelemetryWrapper.openHomescreenShortcutEvent();
+            }
```

Comment:
There's a problem here: This telemetry probe is only fired when the app is already running (onNewIntent). And additionally we send the telemetry probe for the VIEW intent.

We could move this into show showBrowserScreen() [where all incoming URLs go through] as an additional else branch:
https://github.com/mozilla-mobile/focus-android/blob/master/app/src/main/java/org/mozilla/focus/activity/MainActivity.java#L230

Btw. you can test telemetry probes locally (and I should really document this somewhere) by setting telemetryEnabled = true [1] and replacing the upload client with a DebugLogClient [2]. The next time you leave the app this should print the telemetry ping to the log instead of uploading it.

[1] https://github.com/mozilla-mobile/focus-android/blob/master/app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.java#L149
[2] https://github.com/mozilla-mobile/focus-android/blob/master/app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.java#L171